include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -219,7 +215,12 @@ pika::future<T> calcTolerance(SizeType i_begin, SizeType i_end, Matrix<const T,
   };
 
   return ex::when_all(std::move(dmax_fut), std::move(zmax_fut)) |
-         di::transform(di::Policy<Backend::MC>(), std::move(tol_fn)) | ex::make_future();
+         di::transform(di::Policy<Backend::MC>(), std::move(tol_fn)) |
+         // TODO: This releases the tiles that are kept in the operation state.
+         // This is a temporary fix and needs to be replaced by a different
+         // adaptor or different lifetime guarantees. This is tracked in
+         // https://github.com/pika-org/pika/issues/479.
```

Comment:
I've updated the comment here with a link to https://github.com/pika-org/pika/issues/479. A corresponding issue on the DLA-Future side is https://github.com/eth-cscs/DLA-Future/issues/665.

Until those issues are resolved `ensure_started` acts as a hacky replacement for the same purpose and I won't change it in this PR.

Reply:
I don't completely understand the issue here, there are no tiles involved in the call to `di::transform()`. Can you explain a bit more?

Reply:
It maybe doesn't look like it, but the tiles come from `maxVectorElement`. The vectors of `shared_future<Tile>` will be kept for too long in the operation state unless some adaptor explicitly releases the operation states up to this point. `ensure_started` is overkill, but gets the job done (because of how we have implemented it in pika) and is neutral enough to be used like this.